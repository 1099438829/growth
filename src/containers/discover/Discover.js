import React, { Component } from 'react';
import { View, Text, ScrollView } from 'react-native';
import AppStyle from '../../theme/styles';
import MoreItem from '../../components/discover/view/DiscoverMoreItem';
import GridItem from '../../components/discover/view/DiscoverGridItem';
import OnLineRead from '../../components/discover/view/DiscoverOnlineReadItem';
import Launch from '../../components/discover/Launch';

const home1 = require('../../../assets/growth-ui/img/home-1.jpg');
const home2 = require('../../../assets/growth-ui/img/home-2.jpg');
const home3 = require('../../../assets/growth-ui/img/home-3.jpg');

class Discover extends Component {
  static componentName = 'Discover';

  render = () => (
    <ScrollView style={AppStyle.discoverParentStyle}>
      <View style={{ height: 300 }}>
        <View style={AppStyle.discoverTitle}>
          <Text style={AppStyle.discoverTextColor}>在线资源</Text>
        </View>
        <View style={AppStyle.discoverFirstGrid}>
          <GridItem title="学习路线" iconName="icon-social" position={'left'} onclick={() => { Launch.roadmapList(); }} />
          <GridItem title="练手项目" iconName="explore-1-2" position={'right'} onclick={() => { Launch.projectList(); }} />
        </View>
        <View style={AppStyle.discoverOtherGrid}>
          <GridItem title="工具箱" iconName="explore-1-3" position={'left'} onclick={() => { Launch.toolBox(); }} />
          <GridItem title="在线文章" iconName="explore-1-4" position={'right'} onclick={() => { Launch.articleList(); }} />
        </View>
      </View>
      <View style={{ height: 420 }}>
        <View style={AppStyle.discoverTitle}>
          <Text style={AppStyle.discoverTextColor}>本地资源</Text>
        </View>
        <View style={AppStyle.discoverFirstGrid}>
          <GridItem title="技能测验" iconName="icon-graowth" position={'left'} onclick={() => { Launch.examList(); }} />
          <GridItem title="解决方案" iconName="explore-2-2" position={'right'} onclick={() => { Launch.solution(); }} />
        </View>
        <View style={AppStyle.discoverOtherGrid}>
          <GridItem title="读书路线" iconName="explore-2-3" position={'left'} onclick={() => { Launch.thoughtworksBooks(); }} />
          <GridItem title="书籍列表" iconName="explore-2-4" position={'right'} onclick={() => { Launch.recommendBooks(); }} />
        </View>
        <View style={AppStyle.discoverOtherGrid}>
          <GridItem title="文章列表" iconName="explore-2-5" position={'left'} onclick={() => { Launch.recommendArticles(); }} />
          <GridItem title="待办事项" iconName="explore-2-6" position={'right'} onclick={() => { Launch.todoLists(); }} />
        </View>
      </View>
      <View style={{ height: 340 }}>
        <View style={AppStyle.discoverTitle} >
          <Text style={AppStyle.discoverTextColor}>在线阅读</Text>
        </View>
        <OnLineRead image={home1} title={'《Growth: 全栈增长工程师实战》'} onclick={() => { Launch.chapterList(); }} />
        <OnLineRead image={home2} title={'《Ideabock: 练手项目集》'} top={15} onclick={() => { Launch.chapterList(); }} />
      </View>
      <View style={{ height: 190 }}>
        <View style={AppStyle.discoverTitle} >
          <Text style={AppStyle.discoverTextColor}>更多</Text>
        </View>
        <MoreItem
          image={home3}
          title="你还需要什么功能？"
          titleBackgroundColor="rgba(255,255,255, .5)"
          onclick={() => { Launch.openGitHub('https://github.com/phodal/growth-ng/issues'); }}
        />
      </View>
    </ScrollView>
  );
}

export default Discover;
