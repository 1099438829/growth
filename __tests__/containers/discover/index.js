import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import Discover from '../../../src/containers/discover/Discover';

it('renders correctly', () => {
  const discover = renderer.create(
    <Discover />,
  );
  const discoverJson = discover.toJSON();
  expect(discoverJson).toMatchSnapshot();
});
