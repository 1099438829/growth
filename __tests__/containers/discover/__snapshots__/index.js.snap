// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "paddingLeft": 15,
      "paddingRight": 15,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "height": 300,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#03a9f4",
            }
          }
        >
          在线资源
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#03a9f4",
                  },
                  Object {
                    "fontFamily": "growth",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#03a9f4",
                  "marginTop": 10,
                }
              }
            >
              学习路线
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
                "marginLeft": 15,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#03a9f4",
                  },
                  Object {
                    "fontFamily": "growth",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#03a9f4",
                  "marginTop": 10,
                }
              }
            >
              练手项目
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 2,
            "flexDirection": "row",
            "marginTop": 15,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#03a9f4",
                  },
                  Object {
                    "fontFamily": "growth",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#03a9f4",
                  "marginTop": 10,
                }
              }
            >
              工具箱
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
                "marginLeft": 15,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#03a9f4",
                  },
                  Object {
                    "fontFamily": "growth",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#03a9f4",
                  "marginTop": 10,
                }
              }
            >
              在线文章
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 420,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#03a9f4",
            }
          }
        >
          本地资源
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#03a9f4",
                  },
                  Object {
                    "fontFamily": "growth",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#03a9f4",
                  "marginTop": 10,
                }
              }
            >
              技能测验
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
                "marginLeft": 15,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#03a9f4",
                  },
                  Object {
                    "fontFamily": "growth",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#03a9f4",
                  "marginTop": 10,
                }
              }
            >
              解决方案
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 2,
            "flexDirection": "row",
            "marginTop": 15,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#03a9f4",
                  },
                  Object {
                    "fontFamily": "growth",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#03a9f4",
                  "marginTop": 10,
                }
              }
            >
              读书路线
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
                "marginLeft": 15,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#03a9f4",
                  },
                  Object {
                    "fontFamily": "growth",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#03a9f4",
                  "marginTop": 10,
                }
              }
            >
              书籍列表
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 2,
            "flexDirection": "row",
            "marginTop": 15,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#03a9f4",
                  },
                  Object {
                    "fontFamily": "growth",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#03a9f4",
                  "marginTop": 10,
                }
              }
            >
              文章列表
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
                "marginLeft": 15,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#03a9f4",
                  },
                  Object {
                    "fontFamily": "growth",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#03a9f4",
                  "marginTop": 10,
                }
              }
            >
              待办事项
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 340,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#03a9f4",
            }
          }
        >
          在线阅读
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flex": 1,
              "marginTop": 0,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "flex": 7,
                "resizeMode": "stretch",
                "width": 720,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 3,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              《Growth: 全栈增长工程师实战》
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flex": 1,
              "marginTop": 15,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "flex": 7,
                "resizeMode": "stretch",
                "width": 720,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 3,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              《Ideabock: 练手项目集》
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 190,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#03a9f4",
            }
          }
        >
          更多
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flex": 1,
              "marginTop": 0,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "flex": 7,
                "resizeMode": "stretch",
                "width": 720,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, .5)",
                  "flex": 3,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#03a9f4",
                  }
                }
              >
                你还需要什么功能？
              </Text>
            </View>
          </Image>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
